packages/auth-server/pages/recovery/google/index.vue

```diff
@@ -67,36 +72,57 @@ const stepTitle = computed(() => {
   }
 });
 
+function buildBlindingFactor(): bigint {
+  const randomValues = new Uint8Array(31);
+  crypto.getRandomValues(randomValues);
+  return bytesToBigInt(randomValues);
+}
+
 async function generateProf(): Promise<void> {
```

Comment:
1- Make oauth flow to calculate oidc digest.
2- This allow us to query user's address.
3- Calculate txhash
4- Second oauth flow, this time using txHash to build the nonce.
5- calculte proof.